From 26652efc7ea084bdc38da6f37f19ab46aac46944 Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Sun, 15 Apr 2018 20:56:15 +0200
Subject: [PATCH 1/2] selinux: add domain for snap

Snap needs to read vendor files on treble enabled devices.

This is based on policies written by Eino-Ville Talvala <etalvala@google.com> for Google camera
app.

Change-Id: Ia4f3da3fa20ff894b5b202f6691e8f2c03f06e7f
---
 common/private/seapp_contexts |  1 +
 common/private/snap_app.te    | 33 +++++++++++++++++++++++++++++++++
 2 files changed, 34 insertions(+)
 create mode 100644 common/private/seapp_contexts
 create mode 100644 common/private/snap_app.te

diff --git a/common/private/seapp_contexts b/common/private/seapp_contexts
new file mode 100644
index 0000000..2260302
--- /dev/null
+++ b/common/private/seapp_contexts
@@ -0,0 +1 @@
+user=_app isPrivApp=true seinfo=platform name=org.lineageos.snap domain=snap_app type=app_data_file levelFrom=user
diff --git a/common/private/snap_app.te b/common/private/snap_app.te
new file mode 100644
index 0000000..2dce827
--- /dev/null
+++ b/common/private/snap_app.te
@@ -0,0 +1,33 @@
+type snap_app, domain, coredomain;
+
+app_domain(snap_app)
+net_domain(snap_app)
+
+# Access standard system services
+allow snap_app app_api_service:service_manager find;
+allow snap_app audioserver_service:service_manager find;
+allow snap_app cameraserver_service:service_manager find;
+allow snap_app drmserver_service:service_manager find;
+allow snap_app mediacodec_service:service_manager find;
+allow snap_app mediaextractor_service:service_manager find;
+allow snap_app mediaserver_service:service_manager find;
+allow snap_app mediametrics_service:service_manager find;
+allow snap_app nfc_service:service_manager find;
+allow snap_app surfaceflinger_service:service_manager find;
+
+allow snap_app hidl_token_hwservice:hwservice_manager find;
+
+# Allow to read and execute camera app modules
+allow snap_app vendor_file:file { rx_file_perms };
+
+# Execute libraries from RenderScript cache
+allow snap_app app_data_file:file { rx_file_perms };
+
+# Read memory info
+allow snap_app proc_meminfo:file r_file_perms;
+
+# gdbserver / stack traces
+allow snap_app self:process ptrace;
+
+# Read and write system app data files passed over Binder.
+allow snap_app system_app_data_file:file { read write getattr };
-- 
2.17.1

